libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -656,6 +656,16 @@ void CompilerUtils::convertType(Type const& _typeOnStack, Type const& _targetTyp
 		solAssert(_typeOnStack == _targetType, "Invalid type conversion requested.");
 		break;
 	}
+
+	// Check the conversion result fits in a range.
+	if (targetTypeCategory == Type::Category::Enum)
+	{
+		EnumType const* enumType = dynamic_cast<decltype(enumType)>(&_targetType);
```

Comment:
In these cases, I usually do `EnumType const& enumType = dynamic_cast<EnumType const&>(_targetType)` - it will not throw an InternalCompilerError, but the general exception will also be caught.